engine/src/main/java/org/camunda/bpm/engine/impl/persistence/entity/AuthorizationManager.java

```diff
@@ -514,6 +514,31 @@ public void checkCamundaAdmin() {
     }
   }
 
+  public void checkCamundaAdminOrPermission(Consumer<CommandChecker> permissionCheck) {
```

Comment:
Looking at `checkCamundaAdmin`, we first check if authentication is enabled and needs to be checked. Should we do this here as well as a surrounding condition? It probably works right now nonetheless because `checkCamundaAdmin` won't throw an exception currently when authentication is disabled and therefore we don't throw an exception here in the end. But I believe we shouldn't build on this. 😄